src/main/java/frc/robot/subsystems/TurretSubsystem.java

```diff
@@ -184,24 +185,26 @@ public Command testTurretCommand(double degrees) {
   }
 
   private void aimTurret() {
-    double limelightTX = LimelightHelpers.getTX("limelight");
-    if (limelightTX != 0
-        && LimelightHelpers.getFiducialID("limelight")
-            == ((robotStateManager.getAllianceColor() == AllianceColor.BLUE)
-                ? Constants.TurretConstants.SPEAKER_TAG_ID_BLUE
-                : Constants.TurretConstants.SPEAKER_TAG_ID_RED)) {
-      // X & Rotation
-      setTurretPos(Math.toRadians(limelightTX) + turretPosition);
-
-      // Y & Tilting
-      double limelightTY = LimelightHelpers.getTY("limelight");
-      double distanceToTag = tyToDistanceFromTag(limelightTY);
-      tagDistanceEntry.log(distanceToTag);
-      // TODO: Add vertical tilt and use distance for it
-
-      if (Math.abs(Math.toRadians(limelightTX) + turretPosition)
-          > Math.toRadians(Constants.TurretConstants.MAX_TURRET_ANGLE_DEGREES)) {
-        // TODO: Make turret rotate the drivebase if necessary and driver thinks it's a good idea
+    if (visionSubsystem != null) {
+      int tagID =
+          ((robotStateManager.getAllianceColor() == AllianceColor.BLUE)
+              ? Constants.TurretConstants.SPEAKER_TAG_ID_BLUE
+              : Constants.TurretConstants.SPEAKER_TAG_ID_RED);
+      double limelightTX = visionSubsystem.getTurretYaw(tagID);
+      if (limelightTX != 0) {
+        // X & Rotation
+        setTurretPos(Math.toRadians(limelightTX) + turretPosition);
+
+        // Y & Tilting
+        double limelightTY = visionSubsystem.getTurretPitch(tagID);
+        double distanceToTag = tyToDistanceFromTag(limelightTY);
+        tagDistanceEntry.log(distanceToTag);
+        // TODO: Add vertical tilt and use distance for it
+
+        if (Math.abs(Math.toRadians(limelightTX) + turretPosition)
+            > Math.toRadians(Constants.TurretConstants.MAX_TURRET_ANGLE_DEGREES)) {
```

Comment:
As I understand it, starting from pointing straight forward, the turret can go 110 degrees one direction, 5 degrees in the other (or something like that). Is that correct? This code assumes that the turret can go 110 degrees in either direction.

Reply:
commit 329845d has the changes for an asymmetric turret rotation.  

Looks like that is in TurretVerticalControl branch.  Doesn't look like it is in main, yet